leo_fw/src/leo_fw/test_hw.py

```diff
@@ -0,0 +1,385 @@
+                    return 0
+
+        print(self.OKGREEN + "PASSED" + self.ENDC)
+        return 1
+
+    # pylint: disable=too-many-branches,too-many-statements
+    def test_hw(
+        self,
+        hardware=TestMode.ALL,
+    ):
+
+        write_flush("--> Checking if rosserial node is active.. ")
+
+        if rospy.resolve_name("serial_node") in rosnode.get_node_names():
+            print("YES")
+            serial_node_active = True
+        else:
+            print("NO")
+            serial_node_active = False
+            print(
+                "Rosserial node is not active. "
+                "Will not be able to validate hardware."
+                "Try to restart leo.service or reboot system."
+            )
+            return
+
+        #####################################################
+
+        if serial_node_active:
+            write_flush("--> Trying to determine board type.. ")
+
+            board_type = determine_board()
+
+            if board_type is not None:
+                print("SUCCESS")
+            else:
+                print("FAIL")
+
+        #####################################################
+
+        current_firmware_version = "<unknown>"
+
+        if serial_node_active:
+            write_flush("--> Trying to check the current firmware version.. ")
+
+            current_firmware_version = check_firmware_version()
+
+            if current_firmware_version != "<unknown>":
+                print("SUCCESS")
+            else:
+                print("FAIL")
+
+        #####################################################
+
+        if current_firmware_version == "<unknown>" or board_type is None:
+            print(
+                "Can not determine firmware version or board type. "
+                "Flash firmware and try to rerun the script"
+            )
+            return
+
+        #####################################################
+
+        if serial_node_active:
+            write_flush("--> Initializing ROS node.. ")
+            rospy.init_node("leo_core_validation", anonymous=True)
+            print("OK")
+
+        #####################################################
+
+        print(f"Firmware version: {current_firmware_version}")
+        if board_type == BoardType.CORE2:
+            print(f"Board type: Core2ROS")
+        elif board_type == BoardType.LEOCORE:
+            print(f"Board type: LeoCore")
+
+        #####################################################
+
+        if hardware == TestMode.ALL or hardware == TestMode.BATTERY:
+            write_flush("--> Battery validation.. ")
+            self.check_battery()
+
+        if hardware == TestMode.ALL or hardware == TestMode.IMU:
+            write_flush("--> IMU validation.. ")
+            self.check_imu()
+
+        if (
+            hardware == TestMode.ALL
+            or hardware == TestMode.TORQUE
+            or hardware == TestMode.ENCODER
```

Comment:
To check if a variable is equal to one of many values,combine the values into a tuple and check if the variable is contained `in` it. In this example, it would look like this:
```Python
if (hardware in (TestMode.ALL, TestMode.TORQUE, TestMode.ENCODER)):
```
This way, you don't have to disable the `consider-using-in` pylint warning.